### DISCUSSION TITLE
Prisma connection never connect after pool timeout.

### DISCUSSION THREAD

**Discussion by primary-1** [prisma/prisma#10818] [Q&A] [answered]
```
Database:
t3-small (2 CPU, 4GB)

Server:
t3-small (2 CPU, 4GB)

Prisma setting:
10 connections
60 seconds timeout
No external connection poolers

all services running on AWS
```

we using Prisma and cron system at 9:00 am for some automatic process.
that process uses many resources like CPU and RAM usage.

sometimes, it throws from connection timeout then it never tries to connect the database.

<img width="961" alt="Screen Shot 2021-12-22 at 4 00 16 PM" src="https://user-images.githubusercontent.com/21041471/147049953-1c2c6db7-fa70-4dee-937a-9d0d40b07fad.png">

you can see 12 connections at 8:59 am, but after 2~3 minutes later the number of connections be 2.
It throws Error like this

```
Timed out fetching a new connection from the connection pool. More info: http://pris.ly/d/connection-pool (Current connection pool timeout: 60, connection limit: 10)
```

I thought Prisma must connect the database server after a timeout but it never happened.
So I check every morning and if occur this error, I restart the server.
It is intended logic or not?

Additionally, should I disable timeout seconds? or use to external connection pooler?
Of courses, developers should think about database resources but at least I want to make developers more attention to another service code.

**Comment by janpio** [comment, marked as answer]
> I thought Prisma must connect the database server after a timeout but it never happened.
> So I check every morning and if occur this error, I restart the server.
> It is intended logic or not?

Are you running another request after that? Does it just get the same error message although there should be much less usage?

> Additionally, should I disable timeout seconds? or use to external connection pooler?

What is happening here is that Prisma can not handle a query you are trying to do after 60 seconds. That means that during these 60 seconds all the available connections (10) were busy all the time - either by 10 very slow queries or by too many faster queries that still keep all 10 connections busy.

If you do not care that the cron takes longer than 60 seconds to get a connection and execute the query, you can remove the timeout or set it to a much higher value of course. (This is generally a bad idea in web applications where users are waiting for data to be shown of course - but might be just fine for a cron job.)

**Reply by chanlito** [reply]
Is there a way for Prisma to recover from this?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
